sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -328,7 +328,8 @@ object StructType extends AbstractDataType {
   def apply(fields: Seq[StructField]): StructType = StructType(fields.toArray)
 
   def apply(fields: java.util.List[StructField]): StructType = {
-    StructType(fields.toArray.asInstanceOf[Array[StructField]])
+    import scala.collection.JavaConverters._
+    StructType(fields.asScala)
```

Comment:
`DataTypes.createStructType` works well because it use `fields.toArray(new StructField[0])`. However, I think it's bad to create an empty `StructField` array every time we call that method.
You remind me that we should also update `DataTypes.createStructType` to use this `apply` method.